### PR TITLE
Cleanup metadata update frontend

### PR DESCRIPTION
**Pull request by jgraham**
This might make the `wpt update-expectations` command work better as well.

### CODE REVIEW

**Comment by foolip** [comment]
@jgraham there are failing tests, do you want to take a look at those before I review?

File: tools/wpt/update.py
Comment (foolip) [inline tools/wpt/update.py:15]: Any particular reason only one bit is extracted into a variable? Simpler this way? 

```suggestion
        manifest.load_and_update(
            paths["tests_path"],
            paths["manifest_path"],
            url_base)
```

File: tools/wptrunner/wptrunner/wptcommandline.py
Comment (foolip) [inline tools/wptrunner/wptrunner/wptcommandline.py:682]:
```suggestion
                        help="For all tests that are updated, remove any existing conditions and missing subtests")
```

File: tools/wptrunner/wptrunner/wptcommandline.py
Comment (foolip) [inline tools/wptrunner/wptrunner/wptcommandline.py:687]:
```suggestion
                        help="Update test metadata with expected intermittent statuses.")
```

File: tools/wptrunner/wptrunner/wptcommandline.py
Comment (foolip) [inline tools/wptrunner/wptrunner/wptcommandline.py:689]:
```suggestion
                        help="Remove obsolete intermittent statuses from expected statuses.")
```

File: tools/wptrunner/wptrunner/wptcommandline.py
Comment (foolip) [inline tools/wptrunner/wptrunner/wptcommandline.py:691]:
```suggestion
                        help="Don't remove metadata files that no longer correspond to a test file")
```

File: tools/wptrunner/wptrunner/wptcommandline.py
Comment (foolip) [inline tools/wptrunner/wptrunner/wptcommandline.py:694]:
```suggestion
    # TODO: Should make this required iff run=logfile
```

**Review by foolip — COMMENTED** [review]
Are there any tests for this, that will ensure that `./wpt update-expectations` works and keeps working?

**Comment by jgraham** [comment]
I added an integration test that covers `wpt update-expectations`.

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [inline tools/wpt/tests/test_update_expectations.py:73]: Will the test fail if that isn't the case? If it will spuriously pass, is there anything that can be asserted to avoid the risk of later rename?

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [inline tools/wpt/tests/test_update_expectations.py:96]: Will this test work if MANIFEST.json doesn't exist yet? If not, it was probably created by some other integration test, but can we avoid that dependency?

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [inline tools/wpt/tests/test_update_expectations.py:104]: TIL that `with` doesn't create a scope in Python, so `excinfo` lives on!

**Review by foolip — APPROVED** [review]
A few questions, but no need for another review from me.

File: tools/wpt/tests/test_update_expectations.py
Comment (jgraham) [reply tools/wpt/tests/test_update_expectations.py:73]: It will fail I think because we'll end up skipping the test we can't find in the manifest and so not generating metadata for it, so the later asserts will fail.

File: tools/wpt/tests/test_update_expectations.py
Comment (jgraham) [reply tools/wpt/tests/test_update_expectations.py:96]: Yes, because the frontend updates the manifest which will create it if needed.

File: tools/wpt/tests/test_update_expectations.py
Comment (jgraham) [reply tools/wpt/tests/test_update_expectations.py:104]: Haha. Approximately Python scope rules are "after declaration, until the end of the enclosing class or function". It's kind of like old js, except without the hoisting.

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [reply tools/wpt/tests/test_update_expectations.py:73]: Sounds right!

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [reply tools/wpt/tests/test_update_expectations.py:96]: Ah, I guess that happens in the `manifest_update` call.

File: tools/wpt/tests/test_update_expectations.py
Comment (foolip) [reply tools/wpt/tests/test_update_expectations.py:104]: I should know this well by now, but apparently I don't :)

**Comment by foolip** [comment]
@jgraham I'll leave it to you to merge, in case you want to tweak the commit message.

**Pull request merged**
